cmd/manager/main.go

```diff
@@ -219,6 +223,21 @@ func main() {
 		}
 	}
 
+	client := mgr.GetClient()
+	prometheusRule := &monitoringv1.PrometheusRule{ObjectMeta: metav1.ObjectMeta{Name: "mdc-operator-monitoring", Namespace: operatorNamespace}}
+
+	controllerutil.CreateOrUpdate(ctx, client, prometheusRule, func(ignore k8sruntime.Object) error {
+		reconcilePrometheusRule(prometheusRule)
+		return nil
+	})
+
+	grafanaDashboard := &integreatlyv1alpha1.GrafanaDashboard{ObjectMeta: metav1.ObjectMeta{Name: "mdc-operator-monitoring", Namespace: operatorNamespace}}
+
+	controllerutil.CreateOrUpdate(ctx, client, grafanaDashboard, func(ignore k8sruntime.Object) error {
+		reconcileGrafanaDashboard(grafanaDashboard)
+		return nil
+	})
+
```

Comment:
I'd suggest putting this all inside the above block -- the `if service != nil {` one that begins on L209. If there's no Service & therefore no ServiceMonitor, there's no point in creating a PrometheusRule or GrafanaDashboard either. The most common case that this will happen is when running the operator locally.